cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -2338,6 +1905,128 @@ Slice::JavaGenerator::createOutput()
 void
 Slice::JavaGenerator::validateMetadata(const UnitPtr& u)
 {
-    MetadataVisitor visitor;
-    u->visit(&visitor);
+    map<string, MetadataInfo> knownMetadata;
+
+    // "java:buffer"
+    MetadataInfo bufferInfo;
+    bufferInfo.validOn = {typeid(Sequence)};
+    bufferInfo.acceptedArgumentKind = MetadataArgumentKind::NoArguments;
+    bufferInfo.acceptedContext = MetadataApplicationContext::DefinitionsAndTypeReferences;
+    bufferInfo.extraValidation = [](const MetadataPtr&, const SyntaxTreeBasePtr& p) -> optional<string>
+    {
+        if (auto seq = dynamic_pointer_cast<Sequence>(p))
+        {
+            auto builtin = dynamic_pointer_cast<Builtin>(seq->type());
+            if (!builtin || (builtin->kind() != Builtin::KindByte && builtin->kind() != Builtin::KindShort &&
+                             builtin->kind() != Builtin::KindInt && builtin->kind() != Builtin::KindLong &&
+                             builtin->kind() != Builtin::KindFloat && builtin->kind() != Builtin::KindDouble))
+            {
+                return "the 'java:buffer' metadata can only be applied to sequences of bytes, shorts, ints, longs, "
+                       "floats, or doubles";
+            }
+
+            // This check for 'hasMetadata("java:type")' looks redundant, but is necessary to ensure
+            // that 'java:type' was placed on the _definition_ and not on where the sequence was used.
+            if (seq->hasMetadata("java:type") && seq->hasMetadata("java:buffer"))
+            {
+                return "the 'java:buffer' metadata cannot be used alongside 'java:type' - both change the mapped type "
+                       "of this sequence";
+            }
+            if (seq->hasMetadata("java:serializable"))
+            {
+                return "the 'java:buffer' metadata cannot be used alongside 'java:serializable' - both change the "
+                       "mapped type of this sequence";
+            }
+        }
+        return nullopt;
+    };
+    knownMetadata.emplace("java:buffer", std::move(bufferInfo));
+
+    // "java:getset"
+    MetadataInfo getsetInfo;
+    getsetInfo.validOn = {typeid(ClassDecl), typeid(Slice::Exception), typeid(Struct), typeid(DataMember)};
+    getsetInfo.acceptedArgumentKind = MetadataArgumentKind::NoArguments;
+    knownMetadata.emplace("java:getset", std::move(getsetInfo));
+
+    // "java:implements"
```

Comment:
How does this work?

Without partial (like in C#), how is the user going to implement any of the interface methods?

Reply:
From the docs:

> When used with structs, this metadata can only refer to interfaces without operations. With classes or interfaces, the generated Java interface will be marked as abstract. The code is responsible for registering a [value factory](https://doc.zeroc.com/ice/3.7/the-slice-language/classes/value-factories) if the Slice class is transferred over-the-wire and uses this metadata to implement native Java interfaces.

Reply:
To be clear: even though it's documented, it's not usable and should be removed.